reports/src/main/kotlin/com/jakewharton/diffuse/diff/ComponentDiff.kt

```diff
@@ -71,3 +84,50 @@ internal fun StringBuilder.appendComponentDiff(name: String, diff: ComponentDiff
     )
   }
 }
+
+internal fun FlowContent.appendComponentDiff(name: String, diff: ComponentDiff<*>) {
+  if (diff.changed) {
+    h3 { +"$name:" }
+
+    div {
+      style = "margin-left: 16pt;"
+
+      table {
+        thead {
+          tr {
+            td { +"old" }
+            td { +"new" }
+            td { +"diff" }
+          }
+        }
+        tbody {
+          val diffSize = (diff.added.size - diff.removed.size).toDiffString()
+          val addedSize = diff.added.size.toDiffString(zeroSign = '+')
+          val removedSize = (-diff.removed.size).toDiffString(zeroSign = '-')
+
+          tr {
+            td { +diff.oldCount.toString() }
+            td { +diff.newCount.toString() }
+            td { +"$diffSize ($addedSize $removedSize)" }
+          }
+        }
+      }
+
+      if (diff.added.isNotEmpty()) {
+        br()
+        diff.added.forEach {
+          span { unsafe { raw("+ $it".htmlEncoded) } }
```

Comment:
This is kinda lame. kotlinx.html appears to encode some special chars, but not &Delta; or &rarr;.

It would probably be an improvement to somehow avoid using &rarr; in the CF entity toString implementation, but it's not a huge deal (other than allocating every string again…)

Reply:
Replacing &rarr; with `->` would also obviate the need for this. It's easy to replace &Delta; based on the output type because it's written in `:reports` not in a `Method.toString` in `:formats`.

It's maybe a formatting landmine to not have a generic htmlEncoding mechanism.